modules/azure-gaming/connected-resources/modules/resources.bicep

```diff
@@ -0,0 +1,157 @@
+param keyVaultName string = 'keyVault${uniqueString(resourceGroup().id, subscription().id)}'
+
+@allowed([ 'new', 'existing', 'none' ])
+param newOrExistingPublicIp string = 'none'
+param publicIpName string = 'pubip${uniqueString(resourceGroup().id, subscription().id)}'
+
+@allowed([ 'new', 'existing', 'none' ])
+param newOrExistingCosmosDB string = 'none'
+param cosmosDBName string = 'cosmos${uniqueString(resourceGroup().id, subscription().id)}'
+@description('array of region objects or regions: [{locationName: string, failoverPriority: int, isZoneRedundant: bool}] or [region: string]')
+param secondaryLocations array = []
+
+@allowed([ 'new', 'existing', 'none' ])
+param newOrExistingTrafficManager string = 'none'
+param trafficManagerName string = 'traffic-mp-${uniqueString(resourceGroup().id)}'
+@description('Relative DNS name for the traffic manager profile, must be globally unique.')
+param trafficManagerDnsName string = 'tmp-${uniqueString(resourceGroup().id, subscription().id)}'
+
+@allowed([ 'new', 'existing', 'none' ])
+param newOrExistingKubernetes string = 'none'
+param kubernetesParams object = {
+  name: 'aks-${uniqueString(resourceGroup().id)}'
+  agentPoolCount: 3
+  agentPoolName: 'agentpool'
+  vmSize: 'Standard_D2_v2'
+  assignRole: true
+}
+param assignRole bool = true
+
+param storageSecretName string = ''
+param cassandraSecretName string = ''
+
+var newOrExisting = {
+  new: 'new'
+  existing: 'existing'
+}
+
+var enableKubernetes = newOrExistingKubernetes != 'none'
+var enableKeyVault = newOrExistingKeyVault != 'none'
+var enableComosDB = newOrExistingCosmosDB != 'none'
+var enableStorage = newOrExistingStorageAccount != 'none'
+var enablePublicIP = newOrExistingPublicIp != 'none'
+var enableTrafficManager = newOrExistingTrafficManager != 'none'
+
+
+var noAvailabilityZones = [
+  'northcentralus'
+  'westus'
+  'jioindiawest'
+  'westcentralus'
+  'australiacentral'
+  'australiacentral2'
+  'australiasoutheast'
+  'japanwest'
+  'jioindiacentral'
+  'koreasouth'
+  'southindia'
+  'francesouth'
+  'germanynorth'
+  'norwayeast'
+  'switzerlandwest'
+  'ukwest'
+  'uaecentral'
+  'brazilsoutheast'
+]
+
+module clusterModule 'ContainerService/managedClusters.bicep' = if (enableKubernetes) {
+  name: 'create_cluster-${uniqueString(location, resourceGroup().id, deployment().name)}'
+  params: {
+    name: contains(kubernetesParams, 'name') ? kubernetesParams.name : 'aks-${uniqueString(resourceGroup().id)}'
+    location: location
+    agentPoolCount: contains(kubernetesParams, 'agentPoolCount') ? kubernetesParams.agentPoolCount : 3
+    agentPoolName: contains(kubernetesParams, 'agentPoolName') ? kubernetesParams.agentPoolName : 'agentpool'
+    vmSize: contains(kubernetesParams, 'vmSize') ? kubernetesParams.vmSize : 'Standard_D2_v2'
+    assignRole: assignRole
+    newOrExisting: newOrExisting[newOrExistingKubernetes]
+    isZoneRedundant: isZoneRedundant && !contains(noAvailabilityZones, location)
```

Comment:
As you've written this logic in 3 places, i'd promote it a variable to then reference.

```bicep
var locationHasAzSupport = !contains(noAvailabilityZones, location)
```